### ISSUE TITLE
release on fdroid

### ISSUE THREAD

**Issue by c33s** [eszdman/PhotonCamera#46]
https://f-droid.org/en/contribute/

**Comment by Donkey-Doug** [comment]
https://f-droid.org/en/docs/Inclusion_Policy/

**Comment by echeoquehaii** [comment]
The app can't be released on F-Droid?

**Comment by licaon-kter** [comment]
Where do the libs come from? Where are their sources?

https://github.com/eszdman/PhotonCamera/tree/dev/app/src/main/jniLibs

/LE: fixed link

**Comment by eszdman** [comment]
> Where do the libs come from? Where are their sources?
> 
> https://github.com/eszdman/PhotonCamera/app/src/main/jniLibs

Freedcam eszd_hdr branch

**Comment by licaon-kter** [comment]
> Freedcam eszd_hdr branch

Umm? https://github.com/KillerInk/FreeDcam/branches

**Comment by eszdman** [comment]
> Umm

Yes repo is right, probably he removed this branch, I will write to him to return it. I do not add to my repository due to the fact that there may be problems with the build on different systems

**Comment by okias** [comment]
Hello! This camera looks amazing. I would like to try it!

Is there currently any blocker (proprietary dependencies or build not working correctly?) or has anyone tried to get it in yet?

**Comment by eszdman** [comment]
> Hello! This camera looks amazing. I would like to try it!
> 
> Is there currently any blocker (proprietary dependencies or build not working correctly?) or has anyone tried to get it in yet?

Only 1 library is critical - libaiphoto(app package name whitelist)
libhdrx not always used(it's cpu alignment, gpu is fine, fully java side)

**Comment by microbearsmb** [comment]
Is this problematic library used anymore? It's not in the list on the readme. If not, I'd love to see this app on f-droid! Its UI looks lightyears better than the other camera options there.

**Comment by pablo03v** [comment]
This app is on fdroid by now

**Comment by licaon-kter** [comment]
@pabloscloud no, sadly not (yet)

**Comment by pablo03v** [comment]
> @pabloscloud no, sadly not (yet)

![d93c4ea1-ad05-4a96-b0ac-8d9d150977ae](https://github.com/eszdman/PhotonCamera/assets/93644977/5162bd64-e61e-47b9-ad53-878589e51fdc)

**Comment by okias** [comment]
Thank you to the kind person who did the hard work! :)

**Comment by licaon-kter** [comment]
@pabloscloud the topic is "in fdroid" not "binary apk from github hosted in an fdroid repo" (Izzy?)

**Comment by microbearsmb** [comment]
@licaon-kter is correct, Photon Camera is now on IzzyOnDroid, but not F-Droid itself. Helpful for sure, but if it's put on F-Droid it'll be even better.

**Comment by pablo03v** [comment]
I still think it's weird that it is referred to the official fdroid store as fdroid as if other repos didn't exist, but ok it's no in the official fdroid store I guess

**Comment by licaon-kter** [comment]
@pabloscloud it's not strange at all, it says _"in fdroid"_ meaning _"built from the FOSS source and hosted on f-droid.org"_ and not _"blindly hosting a binary built by someone else and add that to my repos"_

**Comment by pablo03v** [comment]
I know it is that way but I personally find it weird. It's just that I open an F-Droid client and see more apps than those in the F-Droid repo so for me they are "in F-Droid" even tho they are not in their repository. But I see why I am wrong 🙂

**Comment by zecamigo** [comment]
@licaon-kter Even on IzzyOnDroid apparently app is set to debuggable. :/

**Comment by eszdman** [comment]
> Is this problematic library used anymore? It's not in the list on the readme. If not, I'd love to see this app on f-droid! Its UI looks lightyears better than the other camera options there.

I made a special branch for FDroid versions without prebuilt libraries 
@licaon-kter you can check now

**Comment by licaon-kter** [comment]
I still see https://github.com/skgmn/RapidDecoder is pulled from a custom maven repo

we'll need to rebuild those libs I guess?

that 11 years (!!) old lib is the best you can use?

**Comment by eszdman** [comment]
> I still see https://github.com/skgmn/RapidDecoder is pulled from a custom maven repo

It's not used, you also don't need to rebuild anything, check latest commit

**Comment by licaon-kter** [comment]
@eszdman it builds now

ok, so, why does it connect to Github?

also, it does not take any pics 😒 

the frame count thing in the button remains as 5 or 2 at random

but appears to take videos 🤷

**Comment by eszdman** [comment]
> @eszdman it builds now
> 
> ok, so, why does it connect to Github?
> 
> also, it does not take any pics 😒 
> 
> the frame count thing in the button remains as 5 or 2 at random
> 
> but appears to take videos 🤷 

Connection to GitHub is necessary to update text configs, for shooting support of RAW is required, the number of frames depends on the lighting, in the dark there are more frames

**Comment by licaon-kter** [comment]
> Connection to GitHub is necessary to update text configs,

so we'd need to add [NonFreeNet](https://f-droid.org/docs/Anti-Features/#NonFreeNet) antifeature

> for shooting support of RAW is required, 

and the app can't detect if the device has this or not?

> the number of frames depends on the lighting, in the dark there are more frames

sure, but why does it remain at 2 or 5 or whatever after shooting?

**Comment by licaon-kter** [comment]
> https://github.com/eszdman/PhotonCamera/blob/dev/app/SupportedList.txt

my device appears in this list, so, it has what is needed, right?

> https://github.com/eszdman/PhotonCamera/tree/dev/app/specific

these are packaged with the app too?

**Comment by eszdman** [comment]
> > https://github.com/eszdman/PhotonCamera/blob/dev/app/SupportedList.txt
> 
> my device appears in this list, so, it has what is needed, right?
> 
> > https://github.com/eszdman/PhotonCamera/tree/dev/app/specific
> 
> these are packaged with the app too?

The list shows whether there is calibration data for shooting and does not always guarantee full functionality, for example when the required blobs are removed in the custom firmware. 
This data is not included in the application, but it is possible to make it as assets
Perhaps you can't save images because you don't have permission to write files, you can look at DCIM/PhotonCamera/PhotonLog

**Comment by licaon-kter** [comment]
> for example when the required blobs are removed in the custom firmware.

I have the official Xiaomi camera in this custom ROM, just a data point

> This data is not included in the application, but it is possible to make it as assets

that would remove the need for an anti-feature flag, yes (eg. maybe have a "update config" button that does an update if the user chooses?)

> Perhaps you can't save images because you don't have permission to write files, 

I've granted it all files access as requested

> you can look at DCIM/PhotonCamera/PhotonLog

saw the file, will take a closer look

**Comment by beedell-roke** [comment]
> that would remove the need for an anti-feature flag, yes (eg. maybe have a "update config" button that does an update if the user chooses?)

@licaon-kter, I know that I would rather have the configurations auto-update, and consequently retain the anti-feature flag. At worst, perhaps have a toggle to auto-update that is disabled by default.

That's because I can't imagine that many users are going to bother to manually update the configuration (considering that from their perspective it should do it itself) so they'll be using out-of-date configurations.

Really, the best solution would just be to host the files somewhere "free" – not GitHub – but I suppose moving the project even to GitLab or Forgejo, instead, is unrealistic.

**Comment by swedneck** [comment]
Been a year since the last comment now, anyone know how things are going?

**Comment by eszdman** [comment]
> Been a year since the last comment now, anyone know how things are going?

It was fixed a long time ago, but they still don't want to add it.

**Comment by licaon-kter** [comment]
@eszdman while we'd like to include all apps under the sun... the day still has 24h 😄 we need help, a merge request with the recipe that is trouble free (sorry if I don't recall the details) can get merged way faster
